desci-server/src/middleware/authorisation.ts

```diff
@@ -46,8 +46,8 @@ export const ensureWriteNodeAccess = async (req: RequestWithUser, res: Response,
   return next();
 };
 
-export const ensureNodeAccess = async (req: RequestWithUser, res: Response, next: NextFunction) => {
-  const user = req.user;
+export const ensureNodeAccess = async (req: Request, res: Response, next: NextFunction) => {
+  const user = (req as RequestWithUser).user;
```

Comment:
I'd try to pass type variables to the `Request<a, b, c>` generic instead of casting, to make it clear what the function expects

Reply:
`req.user` is not a param, body or query. It's an object passed to the request context

Reply:
also that's not a casting, it's a catch one for uuid to make sure the uuid is passed as either a param, body or query so the middleware can authorise access to the node.

The reason being that different request use this middleware and they pass the uuid in either of the 3 different ways (param, body, query)